File: 14-longest-common-prefix/14-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        
        strs.sort()
        
        first = strs[0]
        last = strs[-1]
        count = 0
        for i in range(len(first)):
            if first[i] == last[i]:
                count+=1
            else:
                break
        
        if count == 0:
            return ""
        
        return first[:count]
            
            
        
        
 
        
        
            
            
        

File: 14-longest-common-prefix/14-longest-common-prefix.cpp
#include <vector>
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(),strs.end());
        string string_first = strs[0];
        string string_last = strs.back();
        int count = 0;
        for (int i =0; i < string_first.size(); i++){
            if (string_first[i] == string_last[i]){
                count++;
            }
            else{
                break;
            }
        }
        if (count == 0){
            return "";
        }
        else{
            return string_first.substr(0, count);
        }
    }
};